pkg/client/cache/fifo.go

```diff
@@ -241,6 +241,17 @@ func (f *FIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (f *FIFO) Resync() error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	for id := range f.items {
+		f.queue = append(f.queue, id)
+	}
+	f.cond.Broadcast()
```

Comment:
The intention of a resync is to force and users of a store to re-asses the state of their world.  In this case if there are items in the queue, then you would force the uses of the fifo to re-examine the contents of the fifo and make sure it jives with what the user of the fifo understands to world to be.

A fifo supports all the methods of a store (Add, Get, Delete, etc), so I don't know we be sure that an item will never be in f.items and not also in f.queue.  f.queue is a list of ids to process by the store user.  Once the user of the store has processed that id, which may not result in a Pop of the item, there would be nothing else to cause the fifo user to re-examine a past processed item.  That is what a resync would do.